lightning/src/offers/invoice.rs

```diff
@@ -1416,6 +1417,30 @@ impl Writeable for InvoiceContents {
 	}
 }
 
+impl AsRef<[u8]> for Bolt12Invoice {
+	fn as_ref(&self) -> &[u8] {
+		&self.bytes
+	}
+}
+
+impl Bech32Encode for Bolt12Invoice {
+	const BECH32_HRP: &'static str = "lni";
```

Comment:
Hmm... we shouldn't introduce a bech32 encoding that isn't in the spec. BOLT12 invoices are only transferred over the wire, so one isn't defined. They also only make sense in the context of a preceding `offer` / `invoice_request` (i.e., you'll never see them in a context where a user would scan it).